bundles/org.openhab.binding.chatgpt/src/main/java/org/openhab/binding/chatgpt/internal/ChatGPTHLIService.java

```diff
@@ -0,0 +1,438 @@
+
+        ChatFunction itemControlFunction = FUNCTIONS.get("items_control");
+        if (itemControlFunction != null) {
+
+            itemControlFunction.setParametersClass(ItemsControl.class);
+
+            itemControlFunction.setExecutor(p -> {
+                ItemsControl parameters = (ItemsControl) p;
+
+                return sendCommand(parameters.getName(), parameters.getState());
+            });
+        }
+
+        logger.debug("ChatGPTHLIService activated");
+    }
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        this.config = new Configuration(config).as(ChatGPTHLIConfiguration.class);
+        logger.debug("ChatGPTHLIService configuration completed");
+        for (Map.Entry<String, Object> entry : config.entrySet()) {
+            logger.debug("Config: {} -> {}", entry.getKey(), entry.getValue());
+        }
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        this.config = new Configuration(config).as(ChatGPTHLIConfiguration.class);
+        logger.debug("ChatGPTHLIService configuration modified");
+    }
+
+    @Override
+    public String getId() {
+        return SERVICE_ID;
+    }
+
+    @Override
+    public Set<String> getSupportedGrammarFormats() {
+        return Set.of();
+    }
+
+    @Override
+    public String interpret(Locale locale, String text) {
+
+        String requestBody = prepareRequestBody(text);
+        if (requestBody == null) {
+            return "Failed to prepare request body";
+        }
+
+        if (thingHandler instanceof ChatGPTHandler chatGPTHandler) {
+
+            String response = chatGPTHandler.sendPrompt(requestBody);
+            return processChatResponse(response);
+
+        }
+
+        return "Failed to interpret text";
+    }
+
+    @Override
+    public Set<Locale> getSupportedLocales() {
+
+        return Set.of();
+    }
+
+    @Override
+    public String getLabel(@Nullable Locale locale) {
+
+        return "ChatGPT Human Language Interpreter";
+    }
+
+    @Override
+    @Nullable
+    public String getGrammar(Locale locale, String format) {
+
+        return "null";
+    }
+
+    @Override
+    public void setThingHandler(ThingHandler handler) {
+        this.thingHandler = handler;
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return thingHandler;
+    }
+
+    @Override
+    public void activate() {
+    }
+
+    private String processChatResponse(@Nullable String response) {
+
+        if (response == null || response.isEmpty()) {
+            return "";
+        }
+
+        logger.debug("Received response: {}", response);
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        ChatResponse chatResponse;
+        try {
+            chatResponse = objectMapper.readValue(response, ChatResponse.class);
+        } catch (JsonProcessingException e) {
+            logger.error("Failed to parse ChatGPT response: {}", e.getMessage(), e);
```

Comment:
debug if you also want to log the stacktrace, warn otherwise.
```suggestion
            logger.debug("Failed to parse ChatGPT response: {}", e.getMessage(), e);
```